vector/src/test/java/im/vector/app/features/pin/lockscreen/fragment/LockScreenViewModelTests.kt

```diff
@@ -260,7 +263,44 @@ class LockScreenViewModelTests {
         events.assertValues(LockScreenViewEvent.AuthError(AuthMethod.BIOMETRICS, exception))
     }
 
-    private fun createViewState(
+    @Test
+    fun `when showBiometricPrompt handles isAuthDisabledError, canUseBiometricAuth becomes false`() = runTest {
+        val viewModel = LockScreenViewModel(createViewState(), pinCodeHelper, biometricHelperFactory, keysMigrator, versionProvider, keyguardManager)
+        val exception = BiometricAuthError(BiometricPrompt.ERROR_LOCKOUT_PERMANENT, "Permanent lockout")
+        coEvery { biometricHelper.authenticate(any<FragmentActivity>()) } throws exception
+
+        val events = viewModel.test().viewEvents
```

Comment:
I see that the class already uses this pattern, not a blocker!  
It can be easier from a glance if the test body is broken down into given/when/then blocks, rather than mixing the assertions and methods under test

```
// given 
val viewModel = LockScreenViewModel(createViewState(), pinCodeHelper, biometricHelperFactory, keysMigrator, versionProvider, keyguardManager)
val test = viewModel.test().
val exception = BiometricAuthError(BiometricPrompt.ERROR_LOCKOUT_PERMANENT, "Permanent lockout")
coEvery { biometricHelper.authenticate(any<FragmentActivity>()) } throws exception

// when
viewModel.handle(LockScreenAction.ShowBiometricPrompt(mockk()))

// then
exception.isAuthDisabledError.shouldBeTrue()
test
.assertEvents(LockScreenViewEvent.AuthError(AuthMethod.BIOMETRICS, exception))
.assertLatestState { !it.canUseBiometricAuth } 

```